### ISSUE TITLE
Follow up on dead code removal in AnalyzerHelpers.CreateCompilationWithAnalyzers

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#41246]
The following code was [removed](https://github.com/dotnet/roslyn/pull/40287/files/ae61a7642efdea4603a6f4ba554a0c0915bb65c5#diff-9669ae7e608e196288da6193aebd3a53) as it does nothing:

```C#
            // Always run diagnostic suppressors.	
            analyzers = AppendDiagnosticSuppressors(analyzers, allAnalyzersAndSuppressors: service.GetDiagnosticAnalyzers(project));
```

@mavasani commented:
> I recall that we had to do this because the passed in analyzers were filtered beforehand to exclude analyzers which do not report any SupportedDiagnostics, which is always true for DiagnosticSuppressors. Can you open a bug on me to investigate this and if that is indeed the case still, restore this code with a unit test?

**Comment by mavasani** [comment]
This was investigated and https://github.com/dotnet/roslyn/pull/41867 adds a unit test to ensure suppressors are executed in all supported BackgroundAnalysisScope values. Closing this out.
